walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -18,30 +18,71 @@
 import java.util.Collection;
 import java.util.Collections;
 
-/**
- * Finds time ranges when meeting could happen in a day assuming given events will be sorted from
- * beginning of the day till the end.
- */
 public final class FindMeetingQuery {
+
+  private Collection<String> requiredAttendees;
+  private Collection<String> optionalAttendees;
+  private long duration;
+  public static final int START = TimeRange.START_OF_DAY;
+  public static final int END = TimeRange.START_OF_DAY;
+
+  /** Finds times when meeting could happen in a day. */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
 
     // Gather information from given meeting request.
-    Collection<String> attendeesRequired = request.getAttendees();
-    long duration = request.getDuration();
+    requiredAttendees = request.getAttendees();
+    optionalAttendees = request.getOptionalAttendees();
+    duration = request.getDuration();
+
+    // Initialize return object holding all free time ranges for attendees.
+    ArrayList<TimeRange> possibleMeetingTimes = new ArrayList<TimeRange>();
+    Collection<Event> optionalAttendeeEvents = getOptionalAttendeeEvents(events, optionalAttendees);
+
+    // No required attendees case.
+    if (requiredAttendees.size() == 0) {
+
+      // Check for optional attendees and count them in if in range
+      if (optionalAttendees.size() > 0) {
+        possibleMeetingTimes.addAll(getRanges(optionalAttendees, optionalAttendeeEvents, duration));
+      } else {
+        possibleMeetingTimes.add(TimeRange.WHOLE_DAY);
+      }
+      return possibleMeetingTimes;
+
+      // Longer than a day case.
+    } else if (duration > 1440) {
```

Comment:
Make numbers like this (>0) a constant to improve readability.

Reply:
Changed!